packages/@react-stately/layout/src/ListLayout.ts

```diff
@@ -187,7 +187,7 @@ export class ListLayout<T> extends Layout<Node<T>> implements KeyboardDelegate {
     let layoutNode = this.buildNode(node, x, y);
     layoutNode.node = node;
 
-    layoutNode.layoutInfo.parentKey = node.parentKey || null;
+    layoutNode.layoutInfo.parentKey = node.parentKey ?? null;
```

Comment:
Main culprit for rows w/ falsy ids not rendering (parentKey in layoutInfo was null and thus messed up the association between the cells and the parent row: https://github.com/adobe/react-spectrum/blob/2815eea077010e5490ff9df3c5d79fa1588ca689/packages/%40react-stately/virtualizer/src/Virtualizer.ts#L796-L801 resulting in the TableRow receiving no children)